capabilities/capabilities.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package capabilities
+
+import (
+	"errors"
+	"regexp"
+)
+
+// Capability that expressess an instance Snappy Capability.
+type Capability struct {
+	// Name that is also used for programmatic access on command line and
+	// at application runtime. This is constrained to [a-z][a-z0-9-]+
```

Comment:
// Name is a key that identifies the capability. It must be unique within its context, which
// may be either a snap or a snappy runtime.